client/extensions/zoninator/state/data-layer/zones/index.js

```diff
@@ -24,8 +33,50 @@ export const requestZonesError = ( { dispatch }, { siteId } ) =>
 export const updateZonesList = ( { dispatch }, { siteId }, { data } ) =>
 	dispatch( updateZones( siteId, data ) );
 
+export const createZone = ( { dispatch }, action ) => {
+	const { data, form, siteId } = action;
+
+	dispatch( startSubmit( form ) );
+	dispatch( removeNotice( 'zoninator-zone-create' ) );
+	dispatch( http( {
+		method: 'POST',
+		path: `/jetpack-blogs/${ siteId }/rest-api/`,
+		query: {
+			body: JSON.stringify( data ),
+			json: true,
+			path: '/zoninator/v1/zones',
+		},
+	}, action ) );
+};
+
+export const announceZoneSaved = ( { dispatch }, { form, siteId }, next, { data } ) => {
+	dispatch( stopSubmit( form ) );
+	dispatch( updateZone( siteId, data ) );
```

Comment:
What does the data look like that comes back from the API? If it's in a format that is somewhat ugly (e.g. non-camel case) or has poor naming, then we'll [map the fields](https://github.com/Automattic/wp-calypso/blob/master/client/extensions/wp-job-manager/state/data-layer/settings/utils.js) to better names.